test/extensions/transport_sockets/tls/rsa_private_key_method_provider.cc

```diff
@@ -0,0 +1,296 @@
+/* Copyright (c) 2014, Google Inc.
+ * Copyright (c) 2019, Intel Corp.
+ *
+ * Permission to use, copy, modify, and/or distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY
+ * SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
+ * OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+ * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE. */
```

Comment:
https://github.com/envoyproxy/envoy/blob/master/source/common/chromium_url/LICENSE is how we've handled this for Chromium URL. It's somewhat nicer to have a standalone license file for tooling purposes. I'm not sure what the implication of this particular license or borrowing from BoringSSL is though; in general we have a BoringSSL dependency, so likely have a compatible license, but reading the BoringSSL license makes my head hurt - https://github.com/google/boringssl/blob/master/LICENSE. IANAL; @caniszczyk do you have thoughts on whether we can borrow code in this way?

Reply:
This is one I'd still like to close out.

Reply:
the BoringSSL license(s) are fine for CNCF, just if you use any code, you have to keep all attributions in the header please

Reply:
I decided to rewrite the offending signing code using lower-level RSA methods. I have a tentative patch already, will test it a bit more before submitting.